### ISSUE TITLE
PageView widget does not work with screen reader after navigating away and back

### ISSUE THREAD

**Issue by rami-a** [flutter/flutter#84636]
The PageView widget becomes not usable in screen reader mode when doing the below steps.

## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the main.dart with the contents of this gist: https://gist.github.com/rami-a/e1905d995c673a1e92f4eaf3aaa88571
3. Run the app on device and enable screen reader (talkback/voiceover)
4. Navigate to the last item in the page view
5. Double tap to navigate to a new route
6. Tap back button to go back
7. Now try to navigate through the page view and you will be stuck on the only visible element and unable to see the other elements.

**Expected results:** 
Being able to use the screen reader to navigate through all of the PageView items

**Actual results:** 
Stuck on the currently visible PageView item and unable to navigate through the rest

See video for details:

https://user-images.githubusercontent.com/2364772/122081386-0d4be700-cdcd-11eb-8f45-740e56cc1772.mp4

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel master, 2.3.0-17.0.pre.366, on macOS 11.3.1 20E241 darwin-x64, locale en-US)
    • Flutter version 2.3.0-17.0.pre.366 at /Users/raboughanem/Development/flutter
    • Upstream repository git@github.com:rami-a/flutter.git
    • Framework revision db930ba32f (14 hours ago), 2021-06-14 18:29:03 -0700
    • Engine revision e1d2d4b264
    • Dart version 2.14.0 (build 2.14.0-210.0.dev)
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @rami-a, thanks for filing the issue. 
I am unable to reproduce the issue on `android` on `iOS` with `master 2.5.0-7.0.pre.129` and `stable 2.2.3`.
Kindly see attached videos. 

Caveat: I am not well-versed in the usage of the accessibility mode on either devices, so if my reproduction is wrong kindly let me know and point me in the right direction. If the reproduction is to your satisfaction and you believe that the issue has been solved, kindly leave a comment and I'll close the issue, or you can do that as well. 

Thank you. 

[android.mp4.zip](https://github.com/flutter/flutter/files/7007785/android.mp4.zip)

[ios.MOV.zip](https://github.com/flutter/flutter/files/7007787/ios.MOV.zip)


<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel master, 2.5.0-7.0.pre.129, on macOS 11.5.1 20G80
    darwin-arm, locale en-GH)
    • Flutter version 2.5.0-7.0.pre.129 at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 29c089c2ed (80 minutes ago), 2021-08-18 01:37:04 -0400
    • Engine revision b13c765675
    • Dart version 2.15.0 (build 2.15.0-24.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.59.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (3 available)
    • M2007J20CG (mobile)    • 5dd3be00                  • android-arm64  •
      Android 11 (API 30)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E • ios            • iOS
      14.6 18F72
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 92.0.4515.131

• No issues found!
```

```console
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.1 20G80 darwin-arm, locale
    en-GH)
    • Flutter version 2.2.3 at /Users/nexus/dev/sdks/flutter
    • Framework revision f4abaa0735 (7 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.59.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (3 available)
    • M2007J20CG (mobile)    • 5dd3be00                  • android-arm64  •
      Android 11 (API 30)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E • ios            • iOS
      14.6
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 92.0.4515.131

• No issues found!
```

</details>

**Comment by rami-a** [comment]
Hmm I just tried to reproduce on the latest master, now I'm seeing something different and worse. It does not even let me scroll through any of the items at all, even before tapping into an item

**Comment by danagbemava-nc** [comment]
That is rather odd. May I know what device you were using for your testing?

**Comment by rami-a** [comment]
Yep, I was using a Pixel 5 running Android 11. Its certainly strange behavior

**Comment by danagbemava-nc** [comment]
Perhaps this could be a device specific issue? I attempted with a `OnePlus Nord` this time around, running `Android 11`, and it seems to work just fine.

[one plus nord.mp4.zip](https://github.com/flutter/flutter/files/7014682/one.plus.nord.mp4.zip)

**Comment by chunhtai** [comment]
duplicates of https://github.com/flutter/flutter/issues/40419 and should be fixed
